core/engine/src/value/inner/nan_boxed.rs

```diff
@@ -453,8 +328,12 @@ const_assert!(f64::from_bits(bits::MASK_STRING).is_nan());
 const_assert!(f64::from_bits(bits::MASK_SYMBOL).is_nan());
 const_assert!(f64::from_bits(bits::MASK_BIGINT).is_nan());
 
-/// A NaN-boxed `[JsValue]`'s inner.
-pub(crate) struct NanBoxedValue(pub u64);
+/// A NaN-boxed [`JsValue`]'s inner.
+pub(crate) struct NanBoxedValue {
+    #[cfg(target_pointer_width = "32")]
+    half: u32,
+    ptr: *mut (),
```

Comment:
Have you considered `NonNull<()>` instead?

Reply:
I don't think that would work because we need the null pointer value to represent "0.0" in f64, and using `Option<NonNull<()>>` would make it so that we would have to `transmute` instead.